ddl/partition.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ddl
+
+import (
+	"bytes"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/ast"
+	"github.com/pingcap/tidb/model"
+	"github.com/pingcap/tidb/mysql"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/table"
+)
+
+func buildCreateTablePartitionInfo(ctx sessionctx.Context, d *ddl, s *ast.CreateTableStmt, cols []*table.Column) (*model.PartitionInfo, error) {
+	if s.Partition == nil {
+		return nil, nil
+	}
+	pi := &model.PartitionInfo{
+		Type:   s.Partition.Tp,
+		Enable: ctx.GetSessionVars().EnableTablePartition,
+	}
+	if s.Partition.Expr != nil {
+		buf := new(bytes.Buffer)
+		s.Partition.Expr.Format(buf)
```

Comment:
Why need to format the expr?

Reply:
We can get the `expr` type directly without asserting.